### PR TITLE
AIP-6339: Zodiac service per nb

### PR DESCRIPTION
**Pull request by aaron-arellano**
Support zodiac service and team env variables and adds correct cost labels to kfp pods. Please refer to comment in code as to why we add labels here.  This work is for supporting zodiac service per notebooks and the workflows launched from them. 

This is not tied to CICD, only individual profile notebooks. CICD does not contain zodiac metadata but may be something we visit in the future with further alignment to zkd.

### CODE REVIEW

File: metaflow/metaflow_config.py [hunk 1]
```diff
@@ -156,6 +156,9 @@ def from_conf(name, default=None):
 # For eg, this would look like: "https://<your-kf-cluster-url>/pipeline/#/runs/details/"
 KFP_RUN_URL_PREFIX = from_conf("KFP_RUN_URL_PREFIX", "")
 KFP_MAX_PARALLELISM = int(from_conf("KFP_MAX_PARALLELISM", 10))
+ZODIAC_SERVICE = from_conf("ZODIAC_SERVICE")
+ZODIAC_TEAM = from_conf("ZODIAC_TEAM")
+INDIVIDUAL_NAMESPACE = from_conf("INDIVIDUAL_NAMESPACE", "false")
```
Comment (cloudw) [inline metaflow/metaflow_config.py]: Could you please namespace these variables by prefixing them with "ZILLOW" or any other namespace that makes sense.  For example `ZILLOW_ZODIAC_SERVICE`.

File: metaflow/plugins/kfp/kfp.py [hunk 2]
```diff
@@ -705,21 +720,26 @@ def pipeline_transform(op: ContainerOp):
...
+                # need to be added separately from above as there is no valueFrom/fieldRef from the env
+                # var. leaving as a list format in the event future env variables need to be added without
+                # a fieldRef value_from similar to this env variable.
+                for name, resource in env_vars.items():
+                    op.container.add_env_variable(
+                        V1EnvVar(
+                            name=name,
+                            value=resource,
                         )
+                    )
```
Comment (cloudw) [inline metaflow/plugins/kfp/kfp.py:745]: Looks like `op.container.add_env_variable` is already called on line 715 for part of the `env_vars`. Will line 736 to 742 double add some env vars?  Maybe line 714 to 722 can be removed.

**Review by cloudw — CHANGES_REQUESTED** [review]

File: metaflow/plugins/kfp/kfp.py (hunk 2 above)
Comment (aaron-arellano) [reply metaflow/plugins/kfp/kfp.py:745]: it does not. the var env_vars gets reset at line 725 cause the variables we add in lines 736-742 add those env variables to the container with `name: value`. Lines 714-722 add the env vars set from lines 705-713 with the env var format `name: value_from` which is a different env var format. Both are needed and will not double add env variables.

Comment (cloudw) [reply metaflow/plugins/kfp/kfp.py:745]: Ah I see. Thanks for correcting me!

File: metaflow/metaflow_config.py (hunk 1 above)
Comment (aaron-arellano) [reply metaflow/metaflow_config.py]: I can do that

Comment (cloudw) [reply metaflow/metaflow_config.py]: Thanks! This changes help us avoid name collision, and help identify parts that should be extracted to metaflow extension when we move to argo plugin.

**Review by cloudw — APPROVED** [review]

**Comment by cloudw** [comment]
Looks like some integration tests are failing for 403 errors - you can do a rebase onto `feature/kfp` to pick up fixes from Taleb's commits.

**Pull request merged**